src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -3776,7 +3776,14 @@ void writeString(String value) throws SQLServerException {
             while (bytesCopied < bytesToCopy) {
                 char ch = value.charAt(charsCopied++);
                 valueBytes[bytesCopied++] = (byte) ((ch >> 0) & 0xFF);
-                valueBytes[bytesCopied++] = (byte) ((ch >> 8) & 0xFF);
+
+                if (bytesCopied > bytesToCopy) {
+                    MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_indexOutOfRange"));
+                    Object[] msgArgs = {bytesCopied};
+                    error(form.format(msgArgs), SQLState.DATA_EXCEPTION_NOT_SPECIFIC, DriverError.NOT_SET);
+                }
+                valueBytes[bytesCopied] = (byte) ((ch >> 8) & 0xFF);
+                bytesCopied++;
```

Comment:
This is not an issue, Java char/string are always 2 bytes per character. It's not necessary to do this check.

Reply:
I did this cos adding a try doesn't *fix* the error for Semmle. It's stupid. Ok I changed it